simulation_bridge/src/utils/performance_monitor.py

```diff
@@ -0,0 +1,282 @@
+
+    @property
+    def enabled(self) -> bool:
+        """Return whether performance monitoring is active."""
+        return self._enabled
+
+    @property
+    def history(self) -> List[PerformanceMetrics]:
+        """Return the list of completed metrics."""
+        return list(self._state.metrics_history)
+
+    def get_metric(self, operation_id: str) -> Optional[PerformanceMetrics]:
+        """Return metrics for an ongoing operation."""
+        return self._state.metrics_by_operation_id.get(operation_id)
+
+    def _write_csv_headers(self) -> None:
+        if not self._enabled:
+            return
+        try:
+            with self.paths.csv_path.open("w", newline="", encoding="utf-8") as fh:
+                writer = csv.writer(fh)
+                writer.writerow([
+                    "Timestamp",
+                    "Client ID",
+                    "Client Protocol",
+                    "Operation ID",
+                    "Simulation Type",
+                    "Request Received Time",
+                    "Core Received Input Time",
+                    "Core Sent Input Time",
+                    "Number of Results",
+                    "Result Times",
+                    "Result Sent Times",
+                    "Simulation Request Completed Time",
+                    "CPU Percent",
+                    "Memory RSS (MB)",
+                    "Total Duration",
+                    "Average Result Interval",
+                    "Input Overhead",
+                    "Output Overheads",
+                    "Total Overheads",
+                ])
+        except Exception as exc:
+            logger.error("Failed to write CSV headers: %s", exc)
+            self._enabled = False
+
+    def _update_system_metrics(self, metric: PerformanceMetrics) -> None:
+        if self._state.process:
+            metric.cpu_percent = self._state.process.cpu_percent()
+            metric.memory_rss_mb = (
+                self._state.process.memory_info().rss / (1024 * 1024)
+            )
+
+    def _is_valid_operation(self, operation_id: str) -> bool:
+        return self._enabled and operation_id in self._state.metrics_by_operation_id
+
+    def start_operation(
+        self,
+        operation_id: str,
+        client_id: str = "unknown",
+        protocol: str = "unknown",
+        simulation_type: str = "unknown",
+    ) -> None:
+        if not self._enabled or operation_id in self._state.metrics_by_operation_id:
+            return
+
+        now = time.time()
+        self._state.metrics_by_operation_id[operation_id] = PerformanceMetrics(
+            client_id=client_id,
+            client_protocol=protocol,
+            operation_id=operation_id,
+            simulation_type=simulation_type,
+            timestamp=now,
+            request_received_time=now,
+            core_received_input_time=0.0,
+            core_sent_input_time=0.0,
+        )
+
+    def record_core_received_input(self, operation_id: str) -> None:
```

Comment:
Does the performance_monitor work when two or more simultaneous simulations are going on? Especially with different tuples? The initialization happens fine but all the future recording of events is purely based on operation_id. The idea of changing to tuple is to avoid potential conflicts, right?